packages/mgt/src/components/mgt-agenda/mgt-agenda.ts

```diff
@@ -496,7 +496,45 @@ export class MgtAgenda extends MgtTemplatedComponent {
       return;
     }
 
+    const events = await this.loadEvents();
+    if (events && events.length > 0) {
+      this.events = events;
+    }
+  }
+
+  protected async reload() {
+    this.events = await this.loadEvents();
+    this.render();
+  }
```

Comment:
I'd make the method public and remove the `this.render()` call - otherwise, I think it looks good.

I think it's worth for me to explain how the component rendering works however, it will give you an overview of the lifecycle of the component and explain why `this.render()` is unnecessary. Ideally we should really have this in our contributor guide - sorry about that.

The `loadState` method gets called automatically whenever the component loads or when the provider state changes. As such, we use the `loadState` method to add all the logic to call the graph and setup the data state of the component. 

Now, one really cool part of lit-element is that when a property decorated with the `@property` decorator (as is the `this.events` property) is changed, the component re-renders itself automatically. So we never have to call the render function - the component lifecycle handles that for us. In fact, calling `this.render()` does not do anything - that method is called by lit-element internally - we just need to implement it.

The reloadState method below is actually interesting to note. When a property changes that requires the component to call the graph again (such as starting day), we need to call `loadState` again. To do so, we introduced the `this.requestStateUpdate()` method which handles calling the `loadState` method when possible (`requestStateUpdate` throttles how often `loadState` is called). Before calling `this.requestStateUpdate()`, we set `this.events=null`; to clear the current state and re-render the component with empty state while the data is re-loading.

Hope that makes sense, but happy to go in more detail if that would be helpful.

Reply:
That makes a ton of sense! I was going the `this.requestStateUpdate()` route initially and felt it was not the right approach to "silently" update the `this.events` property. Now everything should be pretty good and it's quite fast! Just a little bit of fiddling around the methods to change where the loading is happening and now we have a new feature! 🚀

Reply:
Beautiful 🍾 

Make the method public and create the docs and this is good to go! ;)

Reply:
Me being too quick on the keyboard, it looks like 😄